framework/Metrics/metrics/RepresentativityFactors.py

```diff
@@ -0,0 +1,113 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 29 2021
+
+@author: Mohammad Abdo (@Jimmy-INL)
+"""
+#for future compatibility with Python 3--------------------------------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+import copy
+#import scipy.spatial.distance as spatialDistance
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from .MetricInterface import MetricInterface
+# from Metrics.metrics import MetricUtilities
+from utils import InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class RepresentativityFactors(MetricInterface):
+  """
+    RepresntativityFactors is the metric class used to quantitatively
+    assess the relativeness of a mock experiment to the target plant.
+  """
+  availScaling ={}
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(RepresentativityFactors, cls).getInputSpecification()
+    actionTypeInput = InputData.parameterInputFactory("actionType", contentType=InputTypes.StringType)
+    inputSpecification.addSub(actionTypeInput)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    # Metric.__init__(self)
+    super().__init__()
+    # The type of given analysis
+    self.actionType                      = None
+    # True indicates the metric needs to be able to handle dynamic data
+    self._dynamicHandling = True
+    # True indicates the metric needs to be able to handle pairwise data
+    self._pairwiseHandling = False
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Method that reads the portion of the xml input that belongs to this specialized class
+      and initialize internal parameters
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = Metric.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    for child in paramInput.subparts:
+      if child.getName() == "actionType":
+        self.order = child.value
```

Comment:
why use "self.order" here, not "self.actionType"?

Reply:
Do I really need this function `_localReadMoreXML` ? I can't remember if I added it. Most of the Metrics do not have it.

Reply:
I have removed it.